src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -278,6 +281,10 @@ public async Task InitializeAsync()
                 {
                     _functionDispatcher.CreateWorkerState(_workerRuntime);
                 }
+                else if (!_environment.IsPlaceholderModeEnabled())
```

Comment:
Can't you internalize all of this logic in the FunctionDispatcher? E.g. replace this entire if/else with `_functionDispatcher.Initialize(functions)` - the dispatcher should already have the environment and can resolve the workerRuntime.

Reply:
done!